### ISSUE TITLE
GET Request with Pydantic List field

### ISSUE THREAD

**Issue by Avish021** [tiangolo/fastapi#2046]
Hi,

I'm new to FastAPI (migrating from Flask) and I'm trying to create a Pydantic Model to my GET route

```
from fastapi import APIRouter,Depends
from pydantic import BaseModel
from typing import Optional,List

router = APIRouter()
class SortModel(BaseModel):
    field:    Optional[str]
    directions: List[str] 

@router.get("/pydanticmodel")
def get_sort(criteria: SortModel = Depends(SortModel)):
    pass #my code for handling this route.....
```

When I'm running  ```curl -X GET http://localhost:XXXX/pydanticmodel?directions=up&directions=asc&field=id
``` I'm getting ```422 Unprocessable Entity: {"detail":[{"loc":["body"],"msg":"field required","type":"value_error.missing"}]}```

But if I'm changing ```directions:List[str]``` -> ```directions: str``` I'm getting ```200 OK``` with directions="asc".

What is the reason that ```str``` works for query param and ```List[str]``` does not? what am I doing wrong?

Thanks.

**Comment by SirTelemak** [comment]
`directions: List[str]` means pydantic should expect array of values on validation, but you can't send `[` and `]` inside query. Unfortunately FastAPI can't work with such case to pass multiple arguments via Depends using pydantic model. Best scenario for you is:
```
class SortModel(BaseModel):

    field:    Optional[str]
    directions: List[str]


def sort_model(field: Optional[str] = Query(None), directions: List[str] = Query([])):
    return SortModel(field=field, directions=directions)


@router.get("/pydanticmodel")
def get_sort(sort: SortModel = Depends(sort_model)):
```

**Comment by ycd** [comment]
@SirTelemak  Yes you can't send an array inside query but you can send a body with **-d** flag, the code works fine when you send a Body with using **-d** flag

```python
In: curl -X GET "http://127.0.0.1:8000/pydanticmodel?field=123"  -d "[\"string\"]"
Out: {"field":"123","directions":["string"]}

```

**Comment by SirTelemak** [comment]
@ycd ye, but it's quite unintuitive and works bad with OpenAPI so I think better way to use like I suggested, so you can use 
`curl -X GET "http://127.0.0.1:8000/pydanticmodel?field=123&directions=value1&directions=value2"`

**Comment by Avish021** [comment]
Hi everybody,

Thank you for your replies. 
Another issue arises since in my real model have like 15 other fields. I would like to avoid passing them as explicit parameters in the sort_model function (since its just repeating the declarations in the model)

Do you know any other way to avoid it?
In addition, thoughts about this work-around? 
```
def add_sort(request:Request):
    d = {}
    for (key, value) in request.query_params._list:
        if key not in d:
            d[key] = value
        else:
            if d[key] is not List:
                current_value = d[key]
                d[key] = [current_value]

            d[key].append( value )

    return SortModel( **d )
```

**Comment by krishnardt** [comment]
Instead of using basemodel, you can use and validate directly through Python class if in case you are getting the data from a form.
class SortModel:
    def __init__(self,
     field= Optional[str]= Form(),
     directions = Optional[str]=Form()):

    self.field = field
    self.directions = directions


@router.get("/pydanticmodel")
def get_sort(criteria: SortModel = Depends()):
    pass #my code for handling this route.....

this will work out for form validation..

**Comment by ycd** [comment]
> @ycd ye, but it's quite unintuitive and works bad with OpenAPI so I think better way to use like I suggested, so you can use
> `curl -X GET "http://127.0.0.1:8000/pydanticmodel?field=123&directions=value1&directions=value2"`

Actually, that's what Swagger suggests for the query.

**Comment by masreplay** [comment]
@tiangolo any help
